CMakeLists.txt

```diff
@@ -284,6 +285,44 @@ if(TOOLCHAIN_CAN_COMPILE_VALE)
     set(HACL_CAN_COMPILE_VALE 1)
 endif()
 
+if(TOOLCHAIN_CAN_COMPILE_AESNI_PCLMUL)
+    add_compile_options(
+        -DHACL_CAN_COMPILE_VEC128
+    )
+    set(HACL_CAN_COMPILE_AESNI_PCLMUL 1)
+
+    # # We make separate compilation units (objects) for each hardware feature
+    list(LENGTH SOURCES_aesni_pclmul SOURCES_AESNI_PCLMUL_LEN)
+
+    if(NOT SOURCES_AESNI_PCLMUL_LEN EQUAL 0)
+        set(HACL_AESNI_PCLMUL_O ON)
+        add_library(hacl_aesni_pclmul OBJECT ${SOURCES_aesni_pclmul})
+        target_include_directories(hacl_aesni_pclmul PRIVATE)
+        if(CMAKE_SYSTEM_PROCESSOR MATCHES "i386|i586|i686|i86pc|ia32|x86_64|amd64|AMD64")
+            add_compile_options(
+                -DHACL_CAN_COMPILE_VEC256
```

Comment:
Same question as for `HACL_CAN_COMPILE_VEC128` above.
Also, you enable both of them without checking for support. If they are really needed here, the entire block should be guarded by both in addition to `TOOLCHAIN_CAN_COMPILE_AESNI_PCLMUL`.